rpcserver.go

```diff
@@ -1120,6 +1121,52 @@ func (r *rpcServer) ListPeers(ctx context.Context,
 	return resp, nil
 }
 
+// ChangePassword changes the password of the currently loaded encrypted wallet,
+// so it must be unlocked first.
+func (r *rpcServer) ChangePassword(ctx context.Context,
+	in *lnrpc.ChangePasswordRequest) (*lnrpc.ChangePasswordResponse, error) {
+
+	// Check macaroon to see if this is allowed.
+	if r.authSvc != nil {
+		if err := macaroons.ValidateMacaroon(ctx, "changepassword",
+			r.authSvc); err != nil {
+			return nil, err
+		}
+	}
+
+	privateWalletPw := in.CurrentPassword
+	publicWalletPw := in.CurrentPassword
+
+	// Check if the current password provided is blank or is the default
+	// wallet password. This most likely means the wallet came from a
+	// --noencryptwallet state. In that case, we set the defaults.
+	if len(in.CurrentPassword) == 0 ||
+		bytes.Compare(in.CurrentPassword, defaultPrivateWalletPw) == 0 {
```

Comment:
I don't think we can check against the private password. Let's say we either remove the length requirement, or make the default password longer than 8 characters. Then a user choosing the default password as her password (without using `--noencryptwallet`), will not be able to change it.

Reply:
Fixed.